app/src/hooks/initialize-agent.ts

```diff
@@ -0,0 +1,262 @@
+            allowCaching: false,
+            cacheDurationInSeconds: 60 * 60 * 24 * 7,
+          },
+        }),
+      }
+
+      logger.info(store.developer.enableProxy && Config.INDY_VDR_PROXY_URL ? 'VDR Proxy enabled' : 'VDR Proxy disabled')
+
+      const newAgent = new Agent(options)
+      const wsTransport = new WsOutboundTransport()
+      const httpTransport = new HttpOutboundTransport()
+
+      newAgent.registerOutboundTransport(wsTransport)
+      newAgent.registerOutboundTransport(httpTransport)
+
+      return newAgent
+    },
+    [walletSecret, store.preferences.walletName, logger, store.developer.enableProxy]
+  )
+
+  const migrateIfRequired = useCallback(
+    async (newAgent: Agent) => {
+      if (!walletSecret?.id || !walletSecret.key) {
+        return
+      }
+
+      // If we haven't migrated to Aries Askar yet, we need to do this before we initialize the agent.
+      if (!store.migration.didMigrateToAskar) {
+        logger.debug('Agent not updated to Aries Askar, updating...')
+
+        await migrateToAskar(walletSecret.id, walletSecret.key, newAgent)
+
+        logger.debug('Successfully finished updating agent to Aries Askar')
+        // Store that we migrated to askar.
+        dispatch({
+          type: DispatchAction.DID_MIGRATE_TO_ASKAR,
+        })
+      }
+    },
+    [walletSecret, store.migration.didMigrateToAskar, dispatch, logger]
+  )
+
+  const warmUpCache = useCallback(
+    async (newAgent: Agent, cachedLedgers?: IndyVdrPoolConfig[]) => {
+      const poolService = newAgent.dependencyManager.resolve(IndyVdrPoolService)
+      if (!cachedLedgers) {
+        // these escapes can be removed once Indy VDR has been upgraded and the patch is no longer needed
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore:next-line
+        await poolService.refreshPoolConnections()
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore:next-line
+        const raw_transactions = await poolService.getAllPoolTransactions()
+        const transactions = raw_transactions
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore:next-line
+          .map((item) => item.value)
+          .map(({ config, transactions }) => ({
+            ...config,
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore:next-line
+            genesisTransactions: transactions.reduce((prev, curr) => {
+              return prev + JSON.stringify(curr)
+            }, ''),
+          }))
+        if (transactions) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          await PersistentStorage.storeValueForKey<any>(BCLocalStorageKeys.GenesisTransactions, {
+            timestamp: moment().toISOString(),
+            transactions,
+          })
+        }
+      }
+
+      credDefs.forEach(async ({ did, id }) => {
```

Comment:
Is forEach await safe? I though that Using await inside a forEach() loop doesn't work as expected because forEach doesn't handle asynchronous functions properly—it doesn't wait for promises to resolve. Instead, you should use a `for...of` loop or `Promise.all()` with `.map()`.

Reply:
This was copied exactly as is from the splash screen in the spirit of keeping all the logic the same. You are right about `await` in `forEach` not behaving the way the original author of this code may have intended but I think if we're going to change it we should do so as part of a separate PR and keep the logic the same for now since it seems to be working ok.